python/foglamp/services/core/api/update.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+""" FogLAMP package updater API support"""
+
+from aiohttp import web
+import uuid
+import asyncio
+import json
+
+# FogLAMP imports
+from foglamp.services.core import server
+from foglamp.common import logger
+
+_LOG_LEVEL_DEBUG = 10
+_logger = logger.setup(__name__, level=_LOG_LEVEL_DEBUG)
+
+__author__ = "Massimiliano Pinto"
+__copyright__ = "Copyright (c) 2018 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_help = """
+    -----------------------------------------
+    | PUT             | /foglamp/update     |
+    -----------------------------------------
+    | no paylod       |                     |
+    -----------------------------------------
+"""
+
+###
+# The update.py is part of the FogLAMP REST API:
+# when called via PUT /foglamp/update it will add the "update task"
+# to scheduler queue for execution.
+#
+# This only happens if the 'FogLAMPUpdater' is found
+# and the scheduled task is enabled.
+#
+
+_FOGLAMP_UPDATE_TASK = "FogLAMPUpdater"
+
+async def update_package(request):
+    """ Queues the execution of FogLAMP package update task
+
+    :Example: curl -X PUT http://localhost:8081/foglamp/update
+    """
+
+    task_found = False
+
+    # Get all the 'Scheduled Tasks'
+    schedule_list = await server.Server.scheduler.get_schedules()
+
+    # Find the FogLAMPUpdater task
+    for schedule_info in schedule_list:
+        if schedule_info.process_name == _FOGLAMP_UPDATE_TASK:
+            task_found = True
+            break;
+
+    # The scheduled task was not set, just return
+    if not task_found:
+        err_message = "FogLAMP update task [" + _FOGLAMP_UPDATE_TASK + "] not configured"
+        _logger.warning(err_message)
+
+        return web.json_response({"status": "Failed", "message": err_message})
+    else:
+        _logger.debug("task " + _FOGLAMP_UPDATE_TASK + " found, ID [" + str(schedule_info.schedule_id) + "]")
+
+    # Status Message prefix for found task
+    status_message =  str(schedule_info.name + " task ID " + str(schedule_info.schedule_id))
+
+    # Now check whether the task is active
+    if schedule_info.enabled == False:
+        status_message += " is not enabled and can't be queued for execution"
```

Comment:
`status_message = "{} is not enabled and can't be queued for execution".format(status_message)`